### PR TITLE
Add MPICH support

### LINKED ISSUE #7: tmpi does not properly work with mpich
I'm on fedora with mpich. See for example what happens with `tmpi 2 python` (spoiler: it reports size 1).

https://asciinema.org/a/HbRzLtMfjNoYaTh0zad4MOoeN

Instead using mpirun i get
```python
➜ mpirun -np 2 python -c "from mpi4py import MPI; print(MPI.COMM_WORLD.Get_size())"
2
2
```
Do you have any suggestion on what might be happening? maybe some env variables are slipping through?

### PR DESCRIPTION
**Pull request by jczhang07**
Thanks to "Raffenetti, Kenneth J." <raffenet@mcs.anl.gov> from the MPICH team for help

### CODE REVIEW

**Comment by jczhang07** [comment]
@raffenet

File: tmpi [hunk 1]
```diff
@@ -55,15 +63,23 @@ if [[ ${1} == runmpi ]] ; then
     # we are being started as one of many processes by mpirun.
     shift
 
+    if [ -n "${mpich}" ]; then
+        mpi_rank=${PMI_ID}
+        mpi_env="-e MPI -e HYDRA -e PMI"
```
Comment (raffenet) [inline tmpi]: I suggest changing `-e HYDRA` to `-e HYD` but otherwise this looks good.

Comment (jczhang07) [reply tmpi]: Fixed.

**Comment by Azrael3000** [comment]
thanks for the pull request. looks good from my side. Let's give @PhilipVinc the chance to test it. Otherwise I'm happy to merge it.

**Comment by jczhang07** [comment]
@PhilipVinc @Azrael3000  Hello, did the PR work? Thanks.

**Comment by PhilipVinc** [comment]
It does work!

**Pull request merged**